### PR TITLE
Handle cancel events (and ESC keypress) by denying the confirmation

### PR DESCRIPTION
**Pull request by gap777**
Addressing https://github.com/RoleModel/turbo-confirm/issues/11

Not sure how to setup the test for this, though... could use help.

### CODE REVIEW

**Comment by OutlawAndy** [comment]
Thanks for this, @gap777.  I will try to look at this later today or tomorrow.

File: src/lib/ConfirmationController.js [hunk 1]
```diff
@@ -59,10 +59,26 @@ export default class ConfirmationController {
...
     // We'll use this to restore the dialog to its original state on teardown.
-    if (!this.#initialContent) this.#initialContent = this.dialogTarget.innerHTML
+    if (!this.#initialContent) {
+      this.#initialContent = this.dialogTarget.innerHTML
+
+      // add a listener for the cancel event on the dialog, to fully deny the confirmation
+      this.dialogTarget.addEventListener('cancel', (event) => {
+        event.preventDefault();
+        this.denyButtons.forEach(element => element.click())
+      });
```
Comment (OutlawAndy) [inline src/lib/ConfirmationController.js:80]: I'm not that familiar with the dialog element, but I'd prefer this live in `#setupListeners` if possible.  Any reason why it couldn't?  I'll comment there with a suggestion.

File: src/lib/ConfirmationController.js [hunk 2]
```diff
@@ -123,12 +139,12 @@ export default class ConfirmationController {
   }
 
   #setupListeners() {
-    this.dialogTarget.querySelectorAll(this.#config.acceptSelector).forEach(element => element.addEventListener('click', this.accept))
-    this.dialogTarget.querySelectorAll(this.#config.denySelector).forEach(element => element.addEventListener('click', this.deny))
+    this.acceptButtons.forEach(element => element.addEventListener('click', this.accept))
+    this.denyButtons.forEach(element => element.addEventListener('click', this.deny))
```
Comment (OutlawAndy) [inline src/lib/ConfirmationController.js:143]:
```suggestion
    this.acceptButtons.forEach(element => element.addEventListener('click', this.accept))
    this.denyButtons.forEach(element => element.addEventListener('click', this.deny))
    this.dialogTarget.addEventListener('cancel', this.deny)
```

File: src/lib/ConfirmationController.js [hunk 3]
```diff
@@ -123,12 +139,12 @@ export default class ConfirmationController {
...
   }
 
   #teardownListeners() {
-    this.dialogTarget.querySelectorAll(this.#config.acceptSelector).forEach(element => element.removeEventListener('click', this.accept))
-    this.dialogTarget.querySelectorAll(this.#config.denySelector).forEach(element => element.removeEventListener('click', this.deny))
+    this.acceptButtons.forEach(element => element.removeEventListener('click', this.accept))
+    this.denyButtons.forEach(element => element.removeEventListener('click', this.deny))
```
Comment (OutlawAndy) [inline src/lib/ConfirmationController.js:148]:
```suggestion
    this.acceptButtons.forEach(element => element.removeEventListener('click', this.accept))
    this.denyButtons.forEach(element => element.removeEventListener('click', this.deny))
    this.dialogTarget.removeEventListener('cancel', this.deny)
```

**Review by OutlawAndy — COMMENTED** [review]
Thanks for submitting this, @gap777.  I do think it's worthwhile for the `dialog` `cancel` event to work as expected here.   I'd just like to maintain the current code organization as much as possible.

As far as testing, I think something like the following would be sufficient.

```js
    it('supports the dialog cancel event', () => {
      const controller = new ConfirmationController()
      const dialog = document.getElementById('confirm')
      controller.perform(event)

      expect(dialog.classList.contains('modal--active')).toBe(true)

      dialog.dispatchEvent(new CustomEvent('cancel'))

      expect(dialog.classList.contains('modal--active')).toBe(false)
    })
```

**Review by OutlawAndy — APPROVED** [review]
Nicely done!  If you wouldn't mind bumping the version number in package.json, I'll tag a release after you squash and merge.  Considering this a patch, unless you disagree?  So `1.1.2` probably?

**Comment by gap777** [comment]
Done

**Pull request merged**